src/tests/orca-jedi/test_interpolator.cc

```diff
@@ -29,105 +30,184 @@ namespace test {
+
+    settings_map["ORCA2_T"].vol_vars = oops::Variables{
+        {oops::Variable{"sea_water_potential_temperature"}}};
+    settings_map["ORCA2_T"].vol_values = std::vector<double>{
+        18.4888916016, missing_value, 18.1592999503,
+        17.9419364929, missing_value, 17.75000288,
+        missing_value, missing_value, missing_value};
   }
 
-  Interpolator interpolator(interpolator_conf, geometry, lats, lons);
-
-  SECTION("test interpolator.apply fails missing variable") {
-    oops::Variables variables{{oops::Variable{"NOTAVARIABLE"}}};
-    std::vector<double> vals(3);
-    std::vector<bool> mask(3, true);
-    EXPECT_THROWS_AS(interpolator.apply(variables, state, mask, vals),
-        eckit::BadParameter);
+  // AMM1 settings
+  {
+    settings_map["AMM1"].nlocs = 3;
+    settings_map["AMM1"].nlevs = 3;
+    settings_map["AMM1"].geometry_config.set("grid name", "../Data/amm1_atlas_grid_spec.yaml");
+    settings_map["AMM1"].geometry_config.set("number levels", settings_map["AMM1"].nlevs);
+
+    std::vector<eckit::LocalConfiguration> nemo_var_mappings(3);
+    nemo_var_mappings[0].set("name", "sea_surface_height_anomaly")
+      .set("nemo field name", "sossheig")
+      .set("model space", "surface");
+    nemo_var_mappings[1].set("name", "mass_concentration_of_chlorophyll_in_sea_water")
+      .set("nemo field name", "CHL")
+      .set("model space", "surface");
+    nemo_var_mappings[2].set("name", "sea_water_potential_temperature")
+      .set("nemo field name", "votemper")
+      .set("model space", "volume");
+    settings_map["AMM1"].geometry_config.set("nemo variables", nemo_var_mappings);
+    eckit::LocalConfiguration interp_conf;
+    interp_conf.set("type", "unstructured-bilinear-lonlat");
+    interp_conf.set("non_linear", "missing-if-all-missing-real32");
+    settings_map["AMM1"].interpolator_config.set("atlas-interpolator", interp_conf);
+
+    std::vector<std::string> state_variables {
+        "sea_surface_height_anomaly",
+        "mass_concentration_of_chlorophyll_in_sea_water",
+        "sea_water_potential_temperature"};
+    settings_map["AMM1"].state_config.set("state variables", state_variables);
+    settings_map["AMM1"].state_config.set("date", "2021-06-30T00:00:00Z");
+    settings_map["AMM1"].state_config.set("nemo field file", "../Data/amm1_nemo.nc");
+    settings_map["AMM1"].state_config.set("variance field file", "../Data/amm1_nemo.nc");
+
+    settings_map["AMM1"].lons = std::vector<double>{-17.5, -6.78, -16.1};
+    settings_map["AMM1"].lats = std::vector<double>{58.16, 58.91, 63.55};
+
+    settings_map["AMM1"].surf_vars = oops::Variables{{
+        oops::Variable{"sea_surface_height_anomaly"},
+        oops::Variable{"mass_concentration_of_chlorophyll_in_sea_water"}}};
+    settings_map["AMM1"].surf_values = std::vector<double>{
+        -0.470139563084, -0.286416769028, -0.433749824762,
+        0.345775008202,  0.83959633112,  2.09327220917};
```

Comment:
Is it worth including a location over land to check masking?

Reply:
I did my best to include some locations over Scotland in the integration ctest for AMM1. They don't show up as entirely missing, because at least one corner of the cell containing the observation is not over land, however if the land mask were not working these observations would match. I believe this is similarly true for the second observation here. But I would need to double check to be sure.

Reply:
No, don't worry. It sounds like the handling of the mask is already being tested adequately.